src/services/arweave/WarpDataProvider.ts

```diff
@@ -149,4 +158,87 @@ export class WarpDataProvider implements SmartweaveDataProvider {
       return error;
     }
   }
+
+  async handleTransactionExecution({
```

Comment:
I'm thinking about this more, do we need this additional function, or can we leverage new types/etc and `writeInteraction` directly 🤔

Reply:
My thought process for this, was that if we put the logic here in this function, we remove logic for this from the transaction page. We pass in the data, tx and contract types, then it runs validations on them to 1: make sure all data is compatible, and 2: make sure that all data is valid.

Reply:
understood - just worried this function is going to get bloated and hard to maintain. switches within switch statements, etc. ideally we do as much validation within the views/pages as possible, and these API calls are succinct/clear. thinking something more similar to the `validationPredicate` pattern we have, passing in required `values` as params within `transactionData` and this logic being more generic/readable

Reply:
removed this, will implement differently